File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/di/AppModule.java
package home.self.beerviewer_mvvm.app_kotlin.di;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import home.self.beerviewer_mvvm.rx.schedulers.BaseSchedulerProvider;
import home.self.beerviewer_mvvm.rx.schedulers.SchedulerProvider;

@Module
public class AppModule {

    @Provides
    @Singleton
    Context provideContext(Application application) {
        return application;
    }

    @Provides
    @Singleton
    BaseSchedulerProvider provideSchedulerProvider() {
        return new SchedulerProvider();
    }
}

File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/di/AppModule.kt
package home.self.beerviewer_mvvm.app_kotlin.di

import android.app.Application
import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import home.self.beerviewer_mvvm.app_kotlin.rx.schedulers.BaseSchedulerProvider
import home.self.beerviewer_mvvm.app_kotlin.rx.schedulers.SchedulerProvider

@Module
class AppModule {

    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application
    }

    @Provides
    @Singleton
    fun provideSchedulerProvider(): BaseSchedulerProvider {
        return SchedulerProvider()
    }
}
